java/ql/src/experimental/Security/CWE/CWE-489/devMode.ql

```diff
@@ -0,0 +1,16 @@
+import java
+import experimental.semmle.code.xml.StrutsXML
+
+bindingset[path]
+predicate isBadPath(string path) {
+  path.matches("%example%") or
+  path.matches("%test%") or
+  path.matches("%demo%")
+}
+
+from ConstantParameter c
+where
+  c.getNameValue() = "struts.devMode" and
+  c.getValueValue() = "true" and
+  not isBadPath(c.getFile().getRelativePath())
+select c
```

Comment:
It would probably also be good to add an [alert message](https://github.com/github/codeql/blob/main/docs/query-metadata-style-guide.md#alert-messages) to explain the problem to the user.